fedimint-api-client/src/api.rs

```diff
@@ -323,6 +319,23 @@ pub trait FederationApiExt: IRawFederationApi {
             request.await
         }
     }
+    async fn request_single_peer_typed<Ret>(
+        &self,
+        timeout: Option<Duration>,
```

Comment:
nit: Do you think that timeout option is useful here? It seems to me that generally handling timeouts at the highest level possible is the way (e.g. if there are two requests don't enforce 1s timeout for each but a total of 2s). Async code can be cancelled, so there is little need for passing timeouts down the call stack.

For consistency I'd try to remove the timeout option from `request_single_peer` later too.

Reply:
Copied it from the existing code, to not expand the change. I support removing it like in @joschisan PR.